src/software/new_geom/rectangle.h

```diff
@@ -101,10 +101,10 @@ class Rectangle : public ConvexPolygon
      * @param amount The amount to expand or shrink the rectangle by on all sides, can be
      * positive or negative
      *
-     * @return bool Whether it was possible to expand/shrink the rectangle by
-     * amount requested, rectangle remains unchanged if impossible to expand/shrink
+     * @return expanded rectangle if it was possible to expand/shrink the rectangle by
+     * amount requested, and return std::nullopt if it wasn't possible
```

Comment:
If we do elect to keep this interface style (return value + side effects) we should definitely note the side effects in this jdoc.

Reply:
see comment above